bin/oneoff/populate_discount_codes.rb

```diff
@@ -0,0 +1,112 @@
+# This script parses a CSV file of discount codes, and uses it to populate our
+# db. The CSV file(s) should not be checked into our repo as they contain secrets.
+# The file(s) are expected to have the following format:
+# 1 header row
+# 1 row for each entry, where a row contains a prefix column and a code column
+# The full code is prefix + code
+require 'securerandom'
+require 'csv'
+require_relative '../../dashboard/config/environment'
+require src_dir 'database'
+
+def populate_csv_data(filename, full_discount, expiration)
+  csv_contents = CSV.read(filename)
+  # skip header row
+  csv_contents.shift
+  csv_contents.each do |line|
+    prefix, code = line
+    CircuitPlaygroundDiscountCode.create!(
+      code: prefix + code,
+      full_discount: full_discount,
+      expiration: expiration
+    )
+  end
+end
+
+full_discount = ARGV[0]
+filename = ARGV[1]
+
+unless full_discount && filename
+  puts 'Expected usage: '
+  puts 'populate_discount_codes <full_discount?> <path_to_csv_file>'
+  exit(-1)
+end
+
+if full_discount == 'true'
+  full_discount = true
+elsif full_discount == 'false'
+  full_discount = false
+else
+  puts 'first argument must be true or false'
+  exit(-1)
+end
+
+populate_csv_data(filename, full_discount, Date.new(2018, 12, 31))
+
+puts 'Complete'
+
+#####
+# Begin debug helper methods
+#
+# What follows is a set of helper methods for discount code functionality that
+# can be copy/pasted into the console. They are designed to be used in non-prod
+# environments
+#####
```

Comment:
Feels a little weird still to have all this debug code live next to our real, slightly related code. Not sure if two separate scripts would make more sense?

Reply:
Agreed. How about a separate file alongside this one with a similar name, e.g. `populate_test_discount_codes`?

Reply:
I think it's fine for this one-off work.